File: mini-commerce/src/main/java/me/hero/minicommerce/item/application/CreateItemService.java
package me.hero.minicommerce.item.application;

import lombok.RequiredArgsConstructor;
import me.hero.minicommerce.item.application.port.in.CreateItemUseCase;
import me.hero.minicommerce.item.application.port.in.dto.CreateItemDto;
import me.hero.minicommerce.item.application.port.out.CreateItemPort;
import me.hero.minicommerce.item.domain.Item;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CreateItemService implements CreateItemUseCase {
  private final CreateItemPort createItemPort;

  @Override
  public CreateItemDto createItem(CreateItemDto itemDto) {
    Item savedItem = createItemPort.saveItem(itemDto.toEntity());
    return new CreateItemDto(savedItem);
  }
}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/application/CreateItemService.kt
package me.hero.minicommerce.item.application

import me.hero.minicommerce.item.application.port.`in`.CreateItemUseCase
import me.hero.minicommerce.item.application.port.`in`.dto.CreateItemDto
import me.hero.minicommerce.item.application.port.out.CreateItemPort
import org.springframework.stereotype.Service

@Service
class CreateItemService(
    private val createItemPort: CreateItemPort
) : CreateItemUseCase {

    override fun createItem(itemDto: CreateItemDto): CreateItemDto {
        val savedItem = createItemPort.saveItem(itemDto.toEntity())
        return CreateItemDto(savedItem)
    }
}